yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -119,7 +131,27 @@ private[yarn] class YarnAllocator(
 
   def getNumExecutorsRunning: Int = numExecutorsRunning
 
-  def getNumExecutorsFailed: Int = numExecutorsFailed
+  def getNumExecutorsFailed: Int = {
+    if(relativeMaxExecutorFailureCheck){
+      getRelevantNumExecutorsFailed
+    } else {
+      numExecutorsFailed.intValue
+    }  
+  }
+
+  /**
+   *  Returns the the relative number of executor failures within the specifid window duration.
+   */
+
+  def getRelevantNumExecutorsFailed : Int = {
+    var currentTime = System.currentTimeMillis / 1000
```

Comment:
Many style problems around here, like the disconnected javadoc, missing spaces around conditions, `var` vs `val`